libsoftwarecontainer/src/container.cpp

```diff
@@ -622,12 +625,23 @@ bool Container::bindMountInContainer(const std::string &pathInHost,
                 log_error() << "Could not create file " << tempPath;
                 return false;
             }
-            m_cleanupHandlers.push_back(new FileCleanUpHandler(tempPath));
         }
     }
 
-    return bindMountCore(pathInHost, pathInContainer, tempPath, readOnly);
+    if (!bindMountCore(pathInHost, pathInContainer, tempPath, readOnly)) {
+        // bindMountInContainer operation is not successful. Clean mounted tempPath.
+        MountCleanUpHandler rollback{tempPath};
+        rollback.clean();
```

Comment:
So we create a `MountCleanUpHandler` with `tempPath` and then instantly clear it of that path?